Storage/Rust/RustLogins.swift

```diff
@@ -142,64 +263,42 @@ public class LoginRecordError: MaybeErrorType {
 }
 
 public class RustLogins {
+    let sqlCipherDatabasePath: String
+    let sqlCipherEncryptionKey: String?
+    let sqlCipherSalt: String?
     let databasePath: String
-    let encryptionKey: String
-    let salt: String
 
     let queue: DispatchQueue
-    let storage: LoginsStorage
+    var storage: LoginStore?
+//    private let queue = DispatchQueue(label: "com.mozilla.logins-storage")
+    let perFieldEncryptionKey: String
 
     private(set) var isOpen = false
 
     private var didAttemptToMoveToBackup = false
-
-    public init(databasePath: String, encryptionKey: String, salt: String) {
+    
+    public init(sqlCipherDatabasePath: String, sqlCipherEncryptionKey: String?, sqlCipherSalt: String?, databasePath: String, perFieldEncryptionKey: String) {
+        self.sqlCipherDatabasePath = sqlCipherDatabasePath
+        self.sqlCipherEncryptionKey = sqlCipherEncryptionKey
+        self.sqlCipherSalt = sqlCipherSalt
         self.databasePath = databasePath
-        self.encryptionKey = encryptionKey
-        self.salt = salt
-
+        self.perFieldEncryptionKey = perFieldEncryptionKey
+        
         queue = DispatchQueue(label: "RustLogins queue: \(databasePath)", attributes: [])
-        storage = LoginsStorage(databasePath: databasePath)
-    }
-
-    // Migrate and return the salt, or create a new salt
-    // Also, in the event of an error, returns a new salt.
-    public static func setupPlaintextHeaderAndGetSalt(databasePath: String, encryptionKey: String) -> String {
-        do {
-            if FileManager.default.fileExists(atPath: databasePath) {
-                let db = LoginsStorage(databasePath: databasePath)
-                let salt = try db.getDbSaltForKey(key: encryptionKey)
-                try db.migrateToPlaintextHeader(key: encryptionKey, salt: salt)
-                return salt
-            }
-        } catch {
-            print(error)
-            Sentry.shared.send(message: "setupPlaintextHeaderAndGetSalt failed", tag: SentryTag.rustLogins, severity: .error, description: error.localizedDescription)
-        }
-        let saltOf32Chars = UUID().uuidString.replacingOccurrences(of: "-", with: "")
-        return saltOf32Chars
     }
 
     // Open the db, and if it fails, it moves the db and creates a new db file and opens it.
     private func open() -> NSError? {
         do {
-            try storage.unlockWithKeyAndSalt(key: encryptionKey, salt: salt)
+            storage = try LoginStore(path: databasePath)
```

Comment:
There will be added functionality here to call canary logic to ensure the per-field key is good and then migrate the database from sqlcipher FYI